Unable to insert a new entity in a table containing name "returning"

## Bug Report

Unable to insert a new entity in a table containing name "returning". The following error observed:
`Statement already includes RETURNING clause`

#### Versions

- Driver: 0.8.6
- Database: Postgres 11.6
- Java: 11
- OS: Mac OS Big Sur

#### Current Behavior

After debugging the guts of the driver I have found that GeneratedValuesUtils#hasReturningClause pattern matcher incorrectly identifies sql query

`INSERT INTO returning_user (id, name, user_details) VALUES ($1, $2, $3)` 

like having returning clause even though it does not have it. 

Looks like RETURNING clause matcher needs to be adjusted for such cases.
<details>
<summary>Stack trace</summary>

```
reactor.core.Exceptions$ErrorCallbackNotImplemented: java.lang.IllegalStateException: Statement already includes RETURNING clause
Caused by: java.lang.IllegalStateException: Statement already includes RETURNING clause
	at io.r2dbc.postgresql.ExtendedQueryPostgresqlStatement.returnGeneratedValues(ExtendedQueryPostgresqlStatement.java:129)
```
</details>
 

#### Expected behavior/code

It's possible to insert records in a table having "returning" in its name.

--- a/src/main/java/io/r2dbc/postgresql/util/GeneratedValuesUtils.java
+++ b/src/main/java/io/r2dbc/postgresql/util/GeneratedValuesUtils.java
@@ -29,7 +29,7 @@ public final class GeneratedValuesUtils {
 
     private static final Pattern INSERT = Pattern.compile(".*INSERT.*", CASE_INSENSITIVE | Pattern.DOTALL);
 
-    private static final Pattern RETURNING = Pattern.compile(".*RETURNING.*", CASE_INSENSITIVE | Pattern.DOTALL);
+    private static final Pattern RETURNING = Pattern.compile(".*\\bRETURNING\\b.*", CASE_INSENSITIVE | Pattern.DOTALL);
 
     private static final Pattern UPDATE = Pattern.compile(".*UPDATE.*", CASE_INSENSITIVE | Pattern.DOTALL);
 

--- a/src/test/java/io/r2dbc/postgresql/util/GeneratedValuesUtilsUnitTests.java
+++ b/src/test/java/io/r2dbc/postgresql/util/GeneratedValuesUtilsUnitTests.java
@@ -52,6 +52,8 @@ final class GeneratedValuesUtilsUnitTests {
     void hasReturning() {
         assertThat(GeneratedValuesUtils.hasReturningClause("test-sql")).isFalse();
         assertThat(GeneratedValuesUtils.hasReturningClause("test-sql RETURNING *")).isTrue();
+        assertThat(GeneratedValuesUtils.hasReturningClause("select * from userReturning")).isFalse();
+        assertThat(GeneratedValuesUtils.hasReturningClause("select * from ReturningUser")).isFalse();
     }
 
     @Test

